### PR TITLE
[Bug] Fix MLFlow import, and setup function that caused tests to fail

### LINKED ISSUE #508: [Bug] Fix MLFlow import
`mlflow` is listed as an optional dependency and thus is not installed unless specified. However, it is imported from in `ml_tools.printing` which throws an error when importing `qadence`.

### PR DESCRIPTION
**Pull request by awennersteen**
Fixes #508

### CODE REVIEW

File: tests/ml_tools/test_logging.py [hunk 1]
```diff
@@ -32,7 +32,7 @@ def dataloader(batch_size: int = 25) -> DataLoader:
     return to_dataloader(x, y, batch_size=batch_size, infinite=True)
 
 
-def setup(model: Module) -> tuple[Callable, Optimizer]:
```
Comment (awennersteen) [inline tests/ml_tools/test_logging.py:35]: This made tests fail on my machine when running tests as it was called on the module. 
I.e. I got failures like
```bash
E       AttributeError: module 'test_logging' has no attribute 'parameters'
```

Comment (Roland-djee) [reply tests/ml_tools/test_logging.py:35]: Not sure this renaming is entirely useful ? `model` is redundant in the name and signature.

**Review by Roland-djee — APPROVED** [review]
Thanks @awennersteen. Not much to say apart that this is looking good.

File: tests/ml_tools/test_logging.py (hunk 1 above)
Comment (awennersteen) [reply tests/ml_tools/test_logging.py:35]: with the naming "model" it failed on my machine. 
Happy to rename it to anything else, but not "setup".

Comment (awennersteen) [reply tests/ml_tools/test_logging.py:35]: For completeness the stack trace
```bash
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> PDB post_mortem (IO-capturing turned off) >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
> /home/awennersteen/work/pasqal/qadence/tests/ml_tools/test_logging.py(38)setup()
-> optimizer = torch.optim.Adam(model.parameters(), lr=0.1)
(Pdb) up
> /home/awennersteen/.local/share/hatch/env/virtual/qadence/kdbNVSwj/qadence/lib/python3.10/site-packages/_pytest/python.py(764)_call_with_optional_argument()
-> func(arg)
(Pdb) up
> /home/awennersteen/.local/share/hatch/env/virtual/qadence/kdbNVSwj/qadence/lib/python3.10/site-packages/_pytest/python.py(630)xunit_setup_module_fixture()
-> _call_with_optional_argument(setup_module, request.module)
(Pdb) up
> /home/awennersteen/.local/share/hatch/env/virtual/qadence/kdbNVSwj/qadence/lib/python3.10/site-packages/_pytest/fixtures.py(910)call_fixture_func()
-> fixture_result = next(generator)
(Pdb) up
> /home/awennersteen/.local/share/hatch/env/virtual/qadence/kdbNVSwj/qadence/lib/python3.10/site-packages/_pytest/fixtures.py(1140)pytest_fixture_setup()
-> result = call_fixture_func(fixturefunc, request, kwargs)
(Pdb) up
> /home/awennersteen/.local/share/hatch/env/virtual/qadence/kdbNVSwj/qadence/lib/python3.10/site-packages/pluggy/_callers.py(102)_multicall()
-> res = hook_impl.function(*args)
(Pdb) up
> /home/awennersteen/.local/share/hatch/env/virtual/qadence/kdbNVSwj/qadence/lib/python3.10/site-packages/_pytest/setuponly.py(36)pytest_fixture_setup()
-> return (yield)
(Pdb) up
> /home/awennersteen/.local/share/hatch/env/virtual/qadence/kdbNVSwj/qadence/lib/python3.10/site-packages/pluggy/_callers.py(121)_multicall()
-> teardown.throw(exception)  # type: ignore[union-attr]
(Pdb) up
> /home/awennersteen/.local/share/hatch/env/virtual/qadence/kdbNVSwj/qadence/lib/python3.10/site-packages/pluggy/_callers.py(138)_multicall()
-> raise exception.with_traceback(exception.__traceback__)
(Pdb) up
> /home/awennersteen/.local/share/hatch/env/virtual/qadence/kdbNVSwj/qadence/lib/python3.10/site-packages/pluggy/_manager.py(119)_hookexec()
-> return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
(Pdb) up
> /home/awennersteen/.local/share/hatch/env/virtual/qadence/kdbNVSwj/qadence/lib/python3.10/site-packages/pluggy/_hooks.py(501)__call__()
-> return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
(Pdb) up
> /home/awennersteen/.local/share/hatch/env/virtual/qadence/kdbNVSwj/qadence/lib/python3.10/site-packages/_pytest/fixtures.py(1087)execute()
-> result = ihook.pytest_fixture_setup(fixturedef=self, request=request)
(Pdb) up
> /home/awennersteen/.local/share/hatch/env/virtual/qadence/kdbNVSwj/qadence/lib/python3.10/site-packages/_pytest/fixtures.py(648)_compute_fixture_value()
-> fixturedef.execute(request=subrequest)
(Pdb) up
> /home/awennersteen/.local/share/hatch/env/virtual/qadence/kdbNVSwj/qadence/lib/python3.10/site-packages/_pytest/fixtures.py(566)_get_active_fixturedef()
-> self._compute_fixture_value(fixturedef)
(Pdb) up
> /home/awennersteen/.local/share/hatch/env/virtual/qadence/kdbNVSwj/qadence/lib/python3.10/site-packages/_pytest/fixtures.py(547)getfixturevalue()
-> fixturedef = self._get_active_fixturedef(argname)
(Pdb) up
> /home/awennersteen/.local/share/hatch/env/virtual/qadence/kdbNVSwj/qadence/lib/python3.10/site-packages/_pytest/fixtures.py(689)_fillfixtures()
-> item.funcargs[argname] = self.getfixturevalue(argname)
(Pdb) up
> /home/awennersteen/.local/share/hatch/env/virtual/qadence/kdbNVSwj/qadence/lib/python3.10/site-packages/_pytest/python.py(1834)setup()
-> self._request._fillfixtures()
(Pdb) up
> /home/awennersteen/.local/share/hatch/env/virtual/qadence/kdbNVSwj/qadence/lib/python3.10/site-packages/_pytest/runner.py(511)setup()
-> col.setup()
(Pdb) up
> /home/awennersteen/.local/share/hatch/env/virtual/qadence/kdbNVSwj/qadence/lib/python3.10/site-packages/_pytest/runner.py(514)setup()
-> raise exc
(Pdb) up
> /home/awennersteen/.local/share/hatch/env/virtual/qadence/kdbNVSwj/qadence/lib/python3.10/site-packages/_pytest/runner.py(158)pytest_runtest_setup()
-> item.session._setupstate.setup(item)
(Pdb) up
> /home/awennersteen/.local/share/hatch/env/virtual/qadence/kdbNVSwj/qadence/lib/python3.10/site-packages/pluggy/_callers.py(102)_multicall()
-> res = hook_impl.function(*args)
(Pdb) up
> /home/awennersteen/.local/share/hatch/env/virtual/qadence/kdbNVSwj/qadence/lib/python3.10/site-packages/_pytest/threadexception.py(63)thread_exception_runtest_hook()
-> yield
(Pdb) up
> /home/awennersteen/.local/share/hatch/env/virtual/qadence/kdbNVSwj/qadence/lib/python3.10/site-packages/_pytest/threadexception.py(82)pytest_runtest_setup()
-> yield from thread_exception_runtest_hook()
(Pdb) up
> /home/awennersteen/.local/share/hatch/env/virtual/qadence/kdbNVSwj/qadence/lib/python3.10/site-packages/pluggy/_callers.py(121)_multicall()
-> teardown.throw(exception)  # type: ignore[union-attr]
(Pdb) up
> /home/awennersteen/.local/share/hatch/env/virtual/qadence/kdbNVSwj/qadence/lib/python3.10/site-packages/_pytest/capture.py(877)pytest_runtest_setup()
-> return (yield)
(Pdb) up
> /home/awennersteen/.local/share/hatch/env/virtual/qadence/kdbNVSwj/qadence/lib/python3.10/site-packages/pluggy/_callers.py(121)_multicall()
-> teardown.throw(exception)  # type: ignore[union-attr]
(Pdb) up
> /home/awennersteen/.local/share/hatch/env/virtual/qadence/kdbNVSwj/qadence/lib/python3.10/site-packages/_pytest/logging.py(822)_runtest_for()
-> yield
(Pdb) up
> /home/awennersteen/.local/share/hatch/env/virtual/qadence/kdbNVSwj/qadence/lib/python3.10/site-packages/_pytest/logging.py(833)pytest_runtest_setup()
-> yield from self._runtest_for(item, "setup")
(Pdb) up
> /home/awennersteen/.local/share/hatch/env/virtual/qadence/kdbNVSwj/qadence/lib/python3.10/site-packages/pluggy/_callers.py(121)_multicall()
-> teardown.throw(exception)  # type: ignore[union-attr]
(Pdb) up
> /home/awennersteen/.local/share/hatch/env/virtual/qadence/kdbNVSwj/qadence/lib/python3.10/site-packages/_pytest/unraisableexception.py(65)unraisable_exception_runtest_hook()
-> yield
(Pdb) up
> /home/awennersteen/.local/share/hatch/env/virtual/qadence/kdbNVSwj/qadence/lib/python3.10/site-packages/_pytest/unraisableexception.py(85)pytest_runtest_setup()
-> yield from unraisable_exception_runtest_hook()
(Pdb) up
> /home/awennersteen/.local/share/hatch/env/virtual/qadence/kdbNVSwj/qadence/lib/python3.10/site-packages/pluggy/_callers.py(121)_multicall()
-> teardown.throw(exception)  # type: ignore[union-attr]
(Pdb) up
> /home/awennersteen/.local/share/hatch/env/virtual/qadence/kdbNVSwj/qadence/lib/python3.10/site-packages/pluggy/_callers.py(138)_multicall()
-> raise exception.with_traceback(exception.__traceback__)
(Pdb) up
> /home/awennersteen/.local/share/hatch/env/virtual/qadence/kdbNVSwj/qadence/lib/python3.10/site-packages/pluggy/_manager.py(119)_hookexec()
-> return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
(Pdb) up
> /home/awennersteen/.local/share/hatch/env/virtual/qadence/kdbNVSwj/qadence/lib/python3.10/site-packages/pluggy/_hooks.py(501)__call__()
-> return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
(Pdb) up
> /home/awennersteen/.local/share/hatch/env/virtual/qadence/kdbNVSwj/qadence/lib/python3.10/site-packages/_pytest/runner.py(263)<lambda>()
-> lambda: ihook(item=item, **kwds), when=when, reraise=reraise
(Pdb) up
> /home/awennersteen/.local/share/hatch/env/virtual/qadence/kdbNVSwj/qadence/lib/python3.10/site-packages/_pytest/runner.py(342)from_call()
-> result: Optional[TResult] = func()
(Pdb) up
*** Oldest frame
```

**Pull request merged**